bears/documentation/DocumentationStyleBear.py

```diff
@@ -46,75 +109,29 @@ def run(self, filename, file, language: str,
-                warning_desc = """
-Missing function description.
-Please set allow_missing_func_desc = True to ignore this warning.
-"""
-            else:
-                warning_desc = 'Documentation does not have correct style.'
-
-            # one empty line shall follow main description (except it's empty
-            # or no annotations follow).
-            if main_description.desc.strip() != '':
-                main_description = main_description._replace(
-                    desc='\n' + main_description.desc.strip() + '\n' *
-                         (1 if len(parsed) == 1 else 2))
-
-            new_metadata = [main_description]
-            for m in metadata:
-                # Split newlines and remove leading and trailing whitespaces.
-                stripped_desc = list(map(str.strip, m.desc.splitlines()))
-
-                if len(stripped_desc) == 0:
-                    # Annotations should be on their own line, though no
-                    # further description follows.
-                    stripped_desc.append('')
-                else:
-                    # Wrap parameter description onto next line if it follows
-                    # annotation directly.
-                    if stripped_desc[0] != '':
-                        stripped_desc.insert(0, '')
-
-                # Indent with 4 spaces.
-                stripped_desc = ('' if line == '' else ' ' * indent_size
-                                 + line for line in stripped_desc)
-
-                new_desc = '\n'.join(stripped_desc)
-
-                # Strip away trailing whitespaces and obsolete newlines (except
-                # one newline which is mandatory).
-                new_desc = new_desc.rstrip() + '\n'
-
-                new_metadata.append(m._replace(desc=new_desc.lstrip(' ')))
+        for doc_comment in self.extraction(language, docstyle):
+            parsed = doc_comment.parse()
+
+            new_metadata = self.fix_parsed_data(parsed,
+                                                allow_missing_func_desc,
+                                                indent_size)
 
             new_comment = DocumentationComment.from_metadata(
                 new_metadata, doc_comment.docstyle_definition,
                 doc_comment.marker, doc_comment.indent, doc_comment.position)
 
             if new_comment != doc_comment:
                 # Something changed, let's apply a result.
-                diff = Diff(file)
-
-                # We need to update old comment positions, as `assemble()`
-                # prepends indentation for first line.
-                old_range = TextRange.from_values(
-                    doc_comment.range.start.line,
-                    1,
-                    doc_comment.range.end.line,
-                    doc_comment.range.end.column)
-                diff.replace(old_range, new_comment.assemble())
+                self.diff_generation(doc_comment, new_comment)
```

Comment:
Related to my [comment](https://github.com/coala/coala/pull/4465#discussion_r128540014) in the other PR: The fact that you store the diff and affected code in a member during this call as a side effect makes it hard for a person reading this code to understand what's going on. I'd prefer it if the results would be returned.